src/Docker.DotNet/Endpoints/ContainerOperations.cs

```diff
@@ -427,5 +445,24 @@ public Task<Stream> GetContainerStatsAsync(string id, ContainerStatsParameters p
             IQueryString queryParameters = new QueryString<ContainerStatsParameters>(parameters);
             return this._client.MakeRequestForStreamAsync(this._client.NoErrorHandlers, HttpMethod.Get, $"containers/{id}/stats", queryParameters, null, cancellationToken);
         }
+
+        public async Task GetContainerStatsAsync(string id, ContainerStatsParameters parameters, CancellationToken cancellationToken, IProgress<ContainerStatsResponse> progress = null)
+        {
+            var responseStream = await GetContainerStatsAsync(id, parameters, cancellationToken);
+
+            using (var reader = new StreamReader(responseStream))
+            {
+                while (responseStream.CanRead && !reader.EndOfStream)
+                {
+                    var line = await reader.ReadLineAsync();
+                    if (progress == null) continue;
+
+                    var report = JsonConvert.DeserializeObject<ContainerStatsResponse>(line);
```

Comment:
Can you use the `this._client.JsonSerializer.DeserializeObject` it makes it a whole lot easier for writing tests to not hard reference JsonConvert